docs/documenting/yaml-style-guide.md

```diff
@@ -0,0 +1,483 @@
+---
+title: "YAML Style Guide"
+---
+
+Additional to our general [documentation standards](/documenting/standards.md),
+also have a set of standards for documenting snippets of YAML. These standards
+described on this page, apply to our all YAML based code across the project,
+with the main focus on documentation.
+
+Our YAML standards provide our end-users with a consistent look, the best
+practices and a uniform approach for solving problems in YAML.
+
+## YAML
+
+This paragraph is about basic YAML usage, and thus not Home Assistant specific.
+
+### Indentation
+
+An indentation of 2 spaces must be used.
+
+```yaml
+# Good
+example:
+  one: 1
+
+# Bad
+example:
+    bad: 2
+```
+
+### Booleans
+
+We should avoid the use of truthy boolean values in YAML. They often trow
+of people new to YAML. Therefore, we only allow the use of `true` and `false`
+as boolean values, in lower case.
+
+This keeps it compatible with the YAML 1.2 specifications as well, since that
+version dropped support for several unquotes truthy booleans (e.g., `y`, `n`,
+`yes`, `no`, `on`, `off` and similar).
+
+  ```yaml
+# Good
+one: true
+two: false
+
+# Bad
+one: True
+two: on
+three: yes
+```
+
+### Comments
+
+Adding comments to blocks of YAML, can really help the reader understand the
+example better.
+
+The indentation level of the comment must match the current indentation level. Preferable is written above the line the comment applies to, otherwise lines
+may become hard to read on smaller displays.
+
+Comments should start with a capital letter and have a space between the
+comment hash `#` and the start of the comment.
+
+```yaml
+# Good
+example:
+  # Comment
+  one: true
+
+# Acceptable, but prefer the above
+example:
+  one: true # Comment
+
+# Bad
+example:
+# Comment
+  one: false
+  #Comment
+  two: false
+  # comment
+  three: false
+```
+
+### Sequences
+
+Sequences in YAML are also known as lists or arrays. In the Home Assistant
+world, we refer to them as lists in end-user documentation. This originates
+from the Python language the Home Assistant core is developed in.
+
+Sequences can be written in two different styles; block and flow style. We
+prefer the use of block style sequences.
+
+#### Block style sequences
+
+Block style sequences need to be indented under the key they belong to.
+
+```yaml
+# Good
+example:
+  - 1
+  - 2
+  - 3
+
+# Bad
+example:
+- 1
+- 2
+- 3
+```
+
+#### Flow style sequences
+
+The use of flow style should be avoided, while simple, short and clean.
+With longer data in it, it becomes long and harder to read quickly.
```

Comment:
```suggestion
The use of flow style should be avoided, although simple, short and clean cases are acceptable.
With longer data in it, it becomes long and harder to read quickly.
```
I think this is what is being said here?

Reply:
Almost :) Tweaked it a bit differently based on another suggestion.